micromanager_gui/_core_widgets/_property_browser.py

```diff
@@ -0,0 +1,189 @@
+from functools import partial
+from typing import Dict, Iterator, Optional, Set, Tuple, cast
+
+from pymmcore_plus import CMMCorePlus, DeviceType
+from qtpy.QtCore import Qt
+from qtpy.QtGui import QColor
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QDialog,
+    QGridLayout,
+    QGroupBox,
+    QHBoxLayout,
+    QLineEdit,
+    QPushButton,
+    QTableWidget,
+    QTableWidgetItem,
+    QVBoxLayout,
+    QWidget,
+)
+
+from .._core import get_core_singleton
+from ._property_widget import PropertyWidget
+
+
+def iter_dev_props(mmc: CMMCorePlus) -> Iterator[Tuple[str, str]]:
+    for dev in mmc.getLoadedDevices():
+        for prop in mmc.getDevicePropertyNames(dev):
+            yield dev, prop
+
+
+class PropertyTable(QTableWidget):
+    def __init__(
+        self, mmcore: Optional[CMMCorePlus] = None, parent: Optional[QWidget] = None
+    ):
+        super().__init__(0, 2, parent=parent)
+        self._mmc = mmcore or get_core_singleton()
+        self._mmc.events.systemConfigurationLoaded.connect(self._rebuild_table)
+        self.destroyed.connect(self._disconnect)
+
+        self.setMinimumWidth(500)
+        self.setHorizontalHeaderLabels(["Property", "Value"])
+        self.setColumnWidth(0, 250)
+        self.horizontalHeader().setStretchLastSection(True)
+        vh = self.verticalHeader()
+        vh.setSectionResizeMode(vh.ResizeMode.Fixed)
+        vh.setDefaultSectionSize(24)
+        vh.setVisible(False)
+        self.setEditTriggers(QTableWidget.EditTrigger.NoEditTriggers)
+        self.setSelectionMode(self.SelectionMode.NoSelection)
+        self.resize(500, 500)
+        self._rebuild_table()
+
+    def _disconnect(self) -> None:
+        self._mmc.events.systemConfigurationLoaded.disconnect(self._rebuild_table)
+
+    def _rebuild_table(self) -> None:
+        self.clearContents()
+        props = list(iter_dev_props(self._mmc))
+        self.setRowCount(len(props))
+        for i, (dev, prop) in enumerate(props):
+            item = QTableWidgetItem(f"{dev}-{prop}")
+            wdg = PropertyWidget(dev, prop, core=self._mmc)
+            self.setItem(i, 0, item)
+            self.setCellWidget(i, 1, wdg)
+            if wdg.isReadOnly():
+                # TODO: make this more theme aware
+                item.setBackground(QColor("#AAA"))
+                wdg.setStyleSheet("QLabel { background-color : #AAA }")
+
+        # TODO: install eventFilter to prevent mouse wheel from scrolling sliders
```

Comment:
This will definitely be nice. It also affects the spinBoxes it seems

Reply:
yeah.  I tried for a while, but it kept failing.  So gave up for later! :)